src/osp/Active/SysVehicle.cpp

```diff
@@ -124,36 +124,30 @@ StatusActivated SysVehicle::activate_sat(ActiveScene &scene,
 
         PrototypePart &proto = *partDepends;
 
-        ActiveEnt partEntity = this->part_instantiate(proto, vehicleEnt);
+        std::vector<MachineDef> machineMapping;
+
+        ActiveEnt partEntity = this->part_instantiate(proto, partBp,
+            vehicleEnt, machineMapping);
+
         vehicleComp.m_parts.push_back(partEntity);
 
         auto& partPart = scene.reg_emplace<ACompPart>(partEntity);
         partPart.m_vehicle = vehicleEnt;
 
         // Part now exists
 
-        // TODO: Deal with blueprint machines instead of prototypes directly
-
-        auto &partMachines = scene.reg_emplace<ACompMachines>(partEntity);
-
-        for (PrototypeMachine& protoMachine : proto.get_machines())
-        {
-            MapSysMachine_t::iterator sysMachine
-                    = scene.system_machine_find(protoMachine.m_type);
-
-            if (!(scene.system_machine_it_valid(sysMachine)))
-            {
-                std::cout << "Machine: " << protoMachine.m_type << " Not found\n";
-                continue;
-            }
-
-            BlueprintMachine blueMach;  // TODO unused so far
-            Machine& machine = sysMachine->second->instantiate(partEntity, protoMachine, blueMach);
-
-            // Add the machine to the part
-            partMachines.m_machines.emplace_back(partEntity, sysMachine);
-
-        }
+        /* Deferred machine instantiation
+        * 
+        * Since machines may depend on the existence of other objects in the
+        * part's hierarchy, machine instantiation must be deferred until the
+        * entire hierarchy exists. For instance, a rocket engine's root
+        * "MachineRocket" machine may depend on the existence of a child node
+        * to receive an "ACompExhaustPlume" component that handles graphical
+        * effects related to the rocket machine, but that child node would not
+        * exist in time for the MachineRocket to be instantiated alongside
+        * the root part entity.
+        */
+        part_instantiate_machines(partEntity, machineMapping, proto, partBp);
```

Comment:
After the part and objects have all been instantiated, the machines are instantiated